File: src/QxRibbonApplicationButton.cpp
﻿/**
 * Copyleft (C) 2023 maminjie <canpool@163.com>
 * SPDX-License-Identifier: MIT
**/
#include "QxRibbonApplicationButton.h"

RibbonApplicationButton::RibbonApplicationButton(QWidget *parent)
    : RibbonApplicationButton(QIcon(), QString(), parent)
{
}

RibbonApplicationButton::RibbonApplicationButton(const QString &text, QWidget *parent)
    : RibbonApplicationButton(QIcon(), text, parent)
{
}

RibbonApplicationButton::RibbonApplicationButton(const QIcon &icon, const QString &text, QWidget *parent)
    : QPushButton(icon, text, parent)
{
    setFocusPolicy(Qt::NoFocus);
    setFlat(true);
    setMinimumWidth(40);
}

File: pyqt/PyRibbon/QxRibbonApplicationButton.py
#!/usr/bin/env python3
# Copyright (c) 2023 maminjie <canpool@163.com>
# SPDX-License-Identifier: MIT

from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QPushButton


class RibbonApplicationButton(QPushButton):

    def __init__(self, *args):
        """
        RibbonApplicationButton(parent=None)
        RibbonApplicationButton(text: str, parent=None)
        RibbonApplicationButton(icon: QIcon, text: str, parent=None)
        """
        super().__init__(*args)
        self.setFocusPolicy(Qt.NoFocus)
        self.setFlat(True)
        self.setMinimumWidth(40)
